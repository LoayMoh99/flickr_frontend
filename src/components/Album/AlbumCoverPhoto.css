.coverPhoto {
  background-size: cover;
  height: 500px;
  width: 100%;
  position: relative;
  background-size: cover;
  background-repeat: no-repeat;
  color: #fff;
  text-shadow: 0 1px 1px #000;
  text-align: center;
}
.avatarPhoto {
  height: 75px;
  width: 75px;
  border-radius: 50%;
  background-size: cover;
}
.overlayAlbum {
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: rgba(0, 0, 0, 0.6);
  z-index: 300;
}
.userName {
  padding-top: 20px;
  color: white;
  font-size: 15px;
}
.countPhotos {
  padding-top: 10px;
  font-size: small;
  font-weight: 200;
}
.spaceAlbumHeader {
  padding-top: 100px;
  padding-right: 70px;
  padding-left: 70px;
}
.spaceTop {
  padding-top: 40px;
}
.editPen {
  padding-left: 850px;
}
.titleEditable {
  font-weight: 300;
  font-size: 3em;
}
.descriptionEditable {
  font-style: italic;
  font-size: 17px;
}
.inputTitleAlbum {
  background-color: transparent;
  border-style: none;
  color: white;
  font-weight: 600;
  font-size: 2em;
}
