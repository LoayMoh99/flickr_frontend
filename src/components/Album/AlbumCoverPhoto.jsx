import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPen } from '@fortawesome/free-solid-svg-icons';
function AlbumCoverPhoto() {
const coverPhoto="https://picsum.photos/500/300?random=1";
const pen =  <FontAwesomeIcon icon={faPen} color="white" size="1x" />;
const avatarUrl="https://picsum.photos/500/300?random=1";
const countPhotos=30;
const userName= "Khadija Swelam";
return (
    <>
    {/* <div className="overlayAlbum"> */}
        {/* <div className="coverPhoto">
            <div className="overlayAlbum" style={{backgroundImage: `url(${coverPhoto})`}}>
                <p>helloooooooooooooooooooooooo
                    oooooooooooooooooooooooooooo
                    ooooooooooooooooooooooooooooo
                </p>
            </div>
        </div> */}
       {/* <img className="img-responsive coverPhoto"src={coverPhoto} alt="image_flickr" /> */}
       {/* <h3>hello</h3> */}
   {/* </div> */}
   {/* <div className="overlayAlbum">
        <img className="img-responsive coverPhoto"src={coverPhoto} alt="image_flickr" />
   </div> */}
   <div className="coverPhoto" style={{backgroundImage: `url(${coverPhoto})`}}>
        <div className="overlayAlbum">
                <div className="spaceAlbumHeader">
                    {/* <span className="spaceTop" >   */}
                        <a href="/#" className="editPen" >   
                                {pen}
                        </a>
                        {/* </span>  */}
                    {/* <input className="inputTitleAlbum" value="Title" type="text" contentEditable="true" ></input> */}
                    <div contentEditable="true" >
                        <div className="titleEditable">
                            Title
                        </div>
                        <div className="descriptionEditable">
                            Click here to enter a description for this album
                        </div>
                    </div>
               </div>
               <h5 className="countPhotos">
                {countPhotos}
               {' '}
                photos</h5>
         <div className="spaceTop">
           <img className="img-responsive avatarPhoto"src={avatarUrl} alt="image_flickr" />
        </div>
        <a href="/#">
            <h5 className="userName"> 
                By:
                {' '}
                {userName}
            </h5>
        </a>
        </div>
   </div>
    </>
  );
}

export default AlbumCoverPhoto;