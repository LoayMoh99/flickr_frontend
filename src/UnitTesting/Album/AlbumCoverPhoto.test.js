import React from 'react';
import ReactDOM from 'react-dom';
import AlbumCoverPhoto from '../../components/Album/AlbumCoverPhoto';
import Enzyme, { shallow, mount } from 'enzyme';
import {render, fireEvent, cleanup} from '@testing-library/react';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() })

afterEach(cleanup)

it('AlbumCoverPhoto renders without crashing', ()=>{
    const div = document.createElement('div');
    ReactDOM.render(<AlbumCoverPhoto 
        coverPhoto = "https://picsum.photos/600/200?random=2"
        title = "Title"
        description = "Description"
        photos = "[{_id:1},{_id:2}]"
        ownerFname ="Khadija"
        ownerLname ="Khaled"
        album_id = "1"
        isUser = "true"
        />, div);

});

it('AlbumCoverPhoto renders correctly enzyme', () => {
    const wrapper = shallow(<AlbumCoverPhoto 
        coverPhoto = "https://picsum.photos/600/200?random=2"
        title = "Title"
        description = "Description"
        photos = "[{_id:1},{_id:2}]"
        ownerFname ="Khadija"
        ownerLname ="Khaled"
        album_id = "1"
        isUser = "true"
        />)
  
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Inputing title updates the state', () => {
    const { getByDisplayValue} = render(<AlbumCoverPhoto 
        coverPhoto = "https://picsum.photos/600/200?random=2"
        title = "Title"
        description = "Description"
        photos = "[{_id:1},{_id:2}]"
        ownerFname ="Khadija"
        ownerLname ="Khaled"
        album_id = "1"
        isUser = "true"
        />);

    expect(getByDisplayValue(/Title/i).value).toBe("Title")

    fireEvent.change(getByDisplayValue("Title"), {target: {value: 'TextTitle' } } )

    expect(getByDisplayValue(/TextTitle/i).value).not.toBe("Title")
 })

 it('Inputing description updates the state', () => {
    const { getByDisplayValue} = render(<AlbumCoverPhoto 
        coverPhoto = "https://picsum.photos/600/200?random=2"
        title = "Title"
        description = "Description"
        photos = "[{_id:1},{_id:2}]"
        ownerFname ="Khadija"
        ownerLname ="Khaled"
        album_id = "1"
        isUser = "true"
        />);

    expect(getByDisplayValue(/Description/i).value).toBe("Description")

    fireEvent.change(getByDisplayValue("Description"), {target: {value: 'TextDescription' } } )

    expect(getByDisplayValue(/TextDescription/i).value).not.toBe("Description")
 })