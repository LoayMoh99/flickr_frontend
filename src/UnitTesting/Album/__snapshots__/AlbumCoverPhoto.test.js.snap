// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumCoverPhoto renders correctly enzyme 1`] = `
<Fragment>
  <div
    className="coverPhoto"
    style={
      Object {
        "backgroundImage": "url(https://picsum.photos/600/200?random=2)",
      }
    }
  >
    <div
      className="overlayAlbum"
    >
      <div
        className="spaceAlbumHeader"
      >
        <div
          className="inputBarAlbumCover"
        >
          <div
            id="inputFormatAlbum"
          >
            <div
              class="form-group"
            >
              <input
                className="form-control"
                onChange={[Function]}
                type="text"
                value="Title"
              />
            </div>
            <div
              class="form-group"
            >
              <textarea
                className="formControlTextarea"
                onChange={[Function]}
                rows="3"
              >
                Description
              </textarea>
            </div>
          </div>
        </div>
      </div>
      <h5
        className="countPhotos"
      >
        17
         
        photos
      </h5>
      <h5
        className="userName"
      >
        <a
          class="userName"
          href="/#"
        >
          By:
           
          Khadija
           
          Khaled
        </a>
      </h5>
    </div>
  </div>
</Fragment>
`;
